app/commands.go

```diff
@@ -55,9 +58,12 @@ type unactivated struct {
 	Level       ui.Level         `help:"Set minimum log level." env:"HERMIT_LOG" default:"info" enum:"trace,debug,info,warn,error,fatal"`
 	GlobalState
 
-	Init       initCmd       `cmd:"" help:"Initialise an environment (idempotent)." group:"env"`
-	Version    versionCmd    `cmd:"" help:"Show version." group:"global"`
-	Validate   validateCmd   `hidden:"" cmd:"" help:"Check a package manifest source for errors." group:"global"`
+	Init     initCmd    `cmd:"" help:"Initialise an environment (idempotent)." group:"env"`
+	Version  versionCmd `cmd:"" help:"Show version." group:"global"`
+	Validate struct {
+		Source validateSourceCmd `default:"withargs" cmd:"" help:"Check a package manifest source for errors." group:"global"`
```

Comment:
This uses a recent contribution from a Kong user to allow default commands with args. The nice thing here is this will keep the command structure backwards compatible.

Reply:
Nice!